### ISSUE TITLE
[1.16.5] Subseasons are being skipped when config file is updated

### ISSUE THREAD

**Issue by Whiskey-Sunshine** [Glitchfiend/SereneSeasons#320]
## Bug Report

As the title states, subseasons are being skipped during the season changes after seasons.toml is updated. This appears to only happen on local worlds, not open to LAN worlds or servers. I ran into this several times playtesting a small modpack I've been building.</br>

### What's the issue you encountered?

On the next season change after a world is saved and closed, the seasons config is changed or even saved again without altering, the next season change will jump straight to the next mid-season, skipping all subseasons in between. It has been most noticeable when using Somnia Awoken (Which does a tick warp rather than a vanilla time skip) but works with or without sleep mechanic altering mods.</br>

### How can the issue be reproduced?

1. Create new world with default config (or update config to desired subseason length)
2. Use the command `/time add 7d` (or subseason length -1 days total, ex 29d if subseason length is 30) to push time to near season change
3. Save and exit world
4. Open and change/update seasons.toml in the sereneseasons config folder
5. Load world and sleep to next day
6. Season will jump forward to next mid-season, skipping all subseasons in between</br>

### Mod Version
SereneSeasons-1.16.5-4.0.1.91-universal
Forge 36.1.2
Additionally tested with: Somnia Awoken 1.16.4-2.0.1</br>

**Comment by Whiskey-Sunshine** [comment]
Forgot to add a screenshot: This world started in subseason 1 (early spring) and by day 91 it is 26 days into mid-winter, when it should only be early summer. The config had been altered or updated a few times since the world was created.

![2021-03-31_20 35 04](https://user-images.githubusercontent.com/30344784/113227591-0b0fcd00-9261-11eb-88c3-f3e60c684485.png)

**Comment by InspectorCaracal** [comment]
I think I got hit by this bug, too, but forgot about the /season command so I'm not 100% sure it skipped to the mid-season - I'll check that tonight when I can get back to my game computer.

TL;DR: This bug appears to be able to happen live in-game regardless of where in the season you are. It may be triggered by: pausing, weather state change, day/night change, or the act of accessing/saving the config.

---

I was using the version for 1.16.5 that was the most recent before this morning, so I think that's 1.16.5-4.0.1.91 and the current Recommended Forge version, 1.16.5 - 36.1.0

* My seasonal config is the defaults, so starting season is 1 (early spring) and subseason length is 8 days
* I spun up a new world and started playing, then waited out the first night after realizing I didn't have any spring-fertile seeds.
* It was either here or after sunrise of day 2 that I opened the fertility.toml file to check which grain seeds would grow in spring.
  * I use Notepad++, not that I think that made a difference. I did not change or save anything.
* I explored a bit, got some spring seeds, and it started raining.
* I started a new mini farm and planted my seeds, started building a small hut and spent the night working on it.
  * The rain stopped again before night fall.
* Morning of Day 3. I went out, chopped down a couple trees, came back to check my plants and discovered that the only one which was growing was a Spring/Summer crop. I was very confused but planted some summer crops instead of spring crops - they worked fine.
  * I'd forgotten about the command to check the current season at this point or I would've just... checked....
* This morning I remembered reading this issue and made the connection between this and accessing my config file and figured I would post.

I'm intending to spin up a test world to try to replicate it on cue tonight, and if I succeed, I'll try the latest version of the mod as well to see if it still does it

**Comment by InspectorCaracal** [comment]
Well, despite my best efforts, I couldn't deliberately replicate what happened to me.

**Comment by InspectorCaracal** [comment]
Just had a season jump happen in my current game and I have a calendar this time so I have more information.

* I did NOT touch the sereneseasons config at all.
* I DID make a couple minor additions to my fertility crop tag files in my personal tweaks/farming compat datapack (and add a recipe that has nothing to do with this mod).
* It did NOT put me in the mid-season; it pushed me from mid summer to late fall.
* As you can probably guess from the tags mention, this is the latest version of the mod, after the config-to-tags switch.

None of those were done while the game was active and I didn't sleep; I just started up the game, went "wait what happened to the grass color, this doesn't look right?" and when I went inside, noticed the calendar said it was late fall.

**Comment by Whiskey-Sunshine** [comment]
I want to mention an update to this issue - I initially thought it had been adjusting the config files, but it seems to happen every time a mod is newly loaded, removed, or updated. Doing some testing with updated mods a couple days ago, the seasons made a jump every time. Went over it several times by moving mods to the disabledmods folder, updating several mods, and adding a couple new ones.

**Comment by mpevner** [comment]
I'm not fully sure how to replicate but I can confirm this happens on multiplayer for me. It seems to follow the above triggers but I have not spent enough time to validate. I do see however the log being spammed with a number of 24,000 season tick skips.

**Comment by Adubbz** [comment]
I could never reproduce this, but in any case, season ticking is now tied to day ticks, so this would almost certainly be fixed now.
